columntype.go

```diff
@@ -0,0 +1,238 @@
+package godatabend
+
+import (
+	"database/sql/driver"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+type ColumnType interface {
+	Desc() *TypeDesc
+	DatabaseTypeName() string
+	Nullable() (bool, bool)
+	ScanType() reflect.Type
+	Parse(s string) (driver.Value, error)
+	Length() (int64, bool)
+	PrecisionScale() (int64, int64, bool)
+}
+
+type columnTypeDefault struct{}
+
+func (columnTypeDefault) Length() (int64, bool) {
+	return 0, false
+}
+
+func (columnTypeDefault) PrecisionScale() (int64, int64, bool) {
+	return 0, 0, false
+}
+
+type unknownColumnType struct {
+	columnTypeDefault
+	dbType string
+	desc   *TypeDesc
+}
+
+func (c unknownColumnType) ScanType() reflect.Type {
+	return reflectTypeString
+}
+
+func (c unknownColumnType) Nullable() (bool, bool) {
+	return false, false
+}
+
+func (c unknownColumnType) DatabaseTypeName() string {
+	return c.dbType
+}
+
+func (c unknownColumnType) Parse(s string) (driver.Value, error) {
+	return s, nil
+}
+
+func (c unknownColumnType) Desc() *TypeDesc {
+	return c.desc
+}
+
+type isNullable bool
+
+func (b isNullable) Nullable() (bool, bool) {
+	return bool(b), true
+}
+
+func (b isNullable) wrapName(s string) string {
+	if bool(b) {
+		return s + " NULL"
+	}
+	return s
+}
+
+func (b isNullable) checkNull(s string) bool {
+	return bool(b) && s == "NULL"
+}
+
+type simpleColumnType struct {
+	dbType    string
+	scanType  reflect.Type
+	nullable  bool
+	parseNull bool
+}
+
+func (*simpleColumnType) PrecisionScale() (int64, int64, bool) {
+	return 0, 0, false
+}
+
+func (c *simpleColumnType) DatabaseTypeName() string {
+	return c.dbType
+}
+
+func (c *simpleColumnType) Nullable() (bool, bool) {
+	return c.nullable, true
+}
+
+func (c *simpleColumnType) Desc() *TypeDesc {
+	return &TypeDesc{Name: c.dbType, Nullable: c.nullable}
+}
+
+func (*simpleColumnType) Length() (int64, bool) {
+	return 0, false
+}
+
+func (c *simpleColumnType) Parse(s string) (driver.Value, error) {
+	if c.nullable && c.parseNull && s == "NULL" {
+		return nil, nil
+	}
+	return s, nil
+}
+
+func (c *simpleColumnType) ScanType() reflect.Type {
+	return c.scanType
+}
+
+type timestampColumnType struct {
+	columnTypeDefault
+	isNullable
+}
+
+func (c timestampColumnType) Parse(s string) (driver.Value, error) {
+	if c.checkNull(s) {
+		return nil, nil
+	}
+	return time.ParseInLocation("2006-01-02 15:04:05.999999", s, time.UTC)
```

Comment:
the timezone should be fetched from settings I think.

Reply:
But we don't have client side timezone, maybe we just use UTC now. 

I checked bendsql it's also using native date

Reply:
i guess it might better to use `time.Parse()` to use the environment's timezone, users can still change the default timezone by themselves if they found the timezone is incorrect, like:

```
    // Set the default timezone to UTC
    loc, err := time.LoadLocation("UTC")
    if err != nil {
        // Handle the error
    }
    time.Local = loc
```

Reply:
I think TIMESTAMP refers to utc time.
Does anyone use the local time zone?

Reply:
We can add a configuration just like [mysql](https://github.com/go-sql-driver/mysql/blob/7403860363ca112af503b4612568c3096fecb466/README.md?plain=1#L293).

Reply:
> We can add a configuration just like [mysql](https://github.com/go-sql-driver/mysql/blob/7403860363ca112af503b4612568c3096fecb466/README.md?plain=1#L293).

yeah, this is pretty nice

Reply:
done